examples/dnn_semantic_segmentation_ex.h

```diff
@@ -34,83 +34,7 @@
 #define DLIB_DNn_SEMANTIC_SEGMENTATION_EX_H_
 
 #include <dlib/dnn.h>
-
-// ----------------------------------------------------------------------------------------
-
-inline bool operator == (const dlib::rgb_pixel& a, const dlib::rgb_pixel& b)
-{
-    return a.red == b.red && a.green == b.green && a.blue == b.blue;
-}
-
-// ----------------------------------------------------------------------------------------
-
-// The PASCAL VOC2012 dataset contains 20 ground-truth classes + background.  Each class
-// is represented using an RGB color value.  We associate each class also to an index in the
-// range [0, 20], used internally by the network.
-
-struct Voc2012class {
```

Comment:
This stuff simply moved to a new header file (used also by the instance segmentation example).